coda-prototype/src/io/FileUtils.ts

```diff
@@ -184,87 +259,80 @@ class FileUtils {
                         deco_timestamp: boolean = eventRow.hasOwnProperty("deco_timestamp"),
                         deco_author: boolean = eventRow.hasOwnProperty("deco_author");
 
-                    // If this parsed row has the minimum information set required to construct an entry in the dataset,
-                    // construct that entry and add it to the dataset.
-                    // TODO: Break this into smaller functions?
-                    if (id && owner && data) {
-                        if (!dataset) {
-                            dataset = new Dataset(); // TODO: Determine whether this check is necessary.
-                        }
+                    console.assert(id && owner && data);
 
-                        let timestampData = timestamp ? eventRow["timestamp"] : "";
+                    let timestampData = timestamp ? eventRow["timestamp"] : "";
 
-                        let isNewEvent = !events.has(eventRow["id"]);
-                        if (isNewEvent) {
-                            nextEvent = new RawEvent(
-                                eventRow["id"], eventRow["owner"], timestampData, eventRow["id"], eventRow["data"]
-                            );
-                            events.set(eventRow["id"], nextEvent);
-                        } else {
-                            nextEvent = events.get(eventRow["id"]);
+                    let isNewEvent = !events.has(eventRow["id"]);
```

Comment:
This looks like data structure fix up code. Can we move it to a separate method in anticipation of wrapping the core data structure in a clean API?

Reply:
In Dataset?

Reply:
Actually, is this the place to do that? Can I tag it as TODO and fix it properly when we do #115